cf-check/dump.c

```diff
@@ -266,72 +251,100 @@ void nice_print_persistent_class(const MDB_val value)
     }
 }
 
-void dump_print_object_line_nice(
-    const MDB_val key, const MDB_val value, const char *file)
+void dump_print_value(
+    const MDB_val key,
+    const MDB_val value,
+    const char *const file,
+    const bool strip_strings,
+    const bool structs)
 {
-    printf("\t");
-    dump_print_json_string_nice(key.mv_data, key.mv_size);
-    printf(": ");
-    if (StringEndsWith(file, "cf_lastseen.lmdb")
-        && StringStartsWith(key.mv_data, "q"))
-    {
-        nice_print_lastseen_quality(value);
-    }
-    else if (StringEndsWith(file, "cf_lock.lmdb"))
-    {
-        nice_print_lock_data(value);
-    }
-    else if (StringEndsWith(file, "cf_observations.lmdb"))
+    bool fallback = false;
+    if (structs)
     {
-        if (StringSafeEqual(key.mv_data, "DATABASE_AGE"))
+        if (StringEndsWith(file, "cf_lastseen.lmdb")
+            && StringStartsWith(key.mv_data, "q"))
+        {
+            print_lastseen_quality(value, strip_strings);
+        }
+        else if (StringEndsWith(file, "cf_lock.lmdb"))
+        {
+            print_lock_data(value, strip_strings);
+        }
+        else if (StringEndsWith(file, "cf_observations.lmdb"))
+        {
+            if (StringSafeEqual(key.mv_data, "DATABASE_AGE"))
+            {
+                assert(sizeof(double) == value.mv_size);
+                const double *const age = value.mv_data;
+                printf("%f", *age);
+            }
+            else
+            {
+                print_averages(value, strip_strings);
+            }
+        }
+        else if (StringEndsWith(file, "history.lmdb"))
         {
-            assert(sizeof(double) == value.mv_size);
-            const double *const age = value.mv_data;
-            printf("%f", *age);
+            print_averages(value, strip_strings);
+        }
+        else if (StringEndsWith(file, "cf_state.lmdb"))
+        {
+            print_persistent_class(value, strip_strings);
         }
         else
         {
-            nice_print_averages(value);
+            fallback = true;
         }
     }
-    else if (StringEndsWith(file, "history.lmdb"))
-    {
-        nice_print_averages(value);
-    }
-    else if (StringEndsWith(file, "cf_state.lmdb"))
-    {
-        nice_print_persistent_class(value);
-    }
-    else
+
+    const bool was_printed = (structs && !fallback);
+    if (!was_printed)
     {
-        dump_print_json_string_nice(value.mv_data, value.mv_size);
+        dump_print_json_string(value.mv_data, value.mv_size, strip_strings);
```

Comment:
I'd prefer `dump_print_json_string(...)` in the `else` branch of the long `if-else if-else` above and in `else` of the `if (structs)`. This `was_printed` is a bit confusing to me.

Reply:
ACK, fixed.